storage/series/series.go

```diff
@@ -433,52 +440,61 @@ func (s *dbSeries) addBlockWithLock(b block.DatabaseBlock) {
 // data in memory during bootstrapping. If that becomes a problem, we could
 // bootstrap in batches, e.g., drain and reset the buffer, drain the streams,
 // then repeat, until len(s.pendingBootstrap) is below a given threshold.
-func (s *dbSeries) Bootstrap(blocks block.DatabaseSeriesBlocks) error {
+func (s *dbSeries) Bootstrap(bootstrappedBlocks block.DatabaseSeriesBlocks) (BootstrapResult, error) {
 	s.Lock()
-	defer s.Unlock()
+	defer s.bootstrapComplete()
```

Comment:
it'd be easier to follow the flow if you were to inline the func declaration here

Reply:
or could alternatively switch it to the `WithLock` thing if you're worried about the extra alloc

```
func (s *dbSeries) Bootstrap(...) (...) {
  s.Lock()
  result, err := s.bootstrapWithLock(...)
  s.bootstrapped = ...
  s.Unlock()
  return result, err
}

func (s *dbSeries) bootstrapWithLock(...) (...) { }

Reply:
Yeah inline would be best.  The lambda will likely get inlined tbh, can check with `go build -gcflags "-m" .` in this package if you want to check.